src/server/network/protocol/protocolgame.cpp

```diff
@@ -2886,17 +2910,39 @@ void ProtocolGame::sendCyclopediaCharacterGeneralStats()
 	// loyalty bonus
 	msg.add<uint16_t>(player->getBaseMagicLevel());
 	msg.add<uint16_t>(player->getMagicLevelPercent() * 100);
+
+	static const uint8_t HardcodedSkillIds[] = { 11, 9, 8, 10, 7, 6, 13 };
 	for (uint8_t i = SKILL_FIRST; i < SKILL_CRITICAL_HIT_CHANCE; ++i)
 	{
-		// check if all clients have the same hardcoded skill ids
-		static const uint8_t HardcodedSkillIds[] = {11, 9, 8, 10, 7, 6, 13};
 		msg.addByte(HardcodedSkillIds[i]);
 		msg.add<uint16_t>(std::min<int32_t>(player->getSkillLevel(i), std::numeric_limits<uint16_t>::max()));
 		msg.add<uint16_t>(player->getBaseSkill(i));
 		// loyalty bonus
 		msg.add<uint16_t>(player->getBaseSkill(i));
 		msg.add<uint16_t>(player->getSkillPercent(i) * 100);
 	}
+
+	// Version 12.70 start (new items magic level modifiers)
+	const Item *weapon = player->getWeapon();
+	if (weapon)
+	{
+		const ItemType &it = Item::items[weapon->getID()];
+		if (it.abilities && it.abilities->elementType != COMBAT_NONE && weapon->getSpecializedMagicLevel(it.abilities->elementType))
+		{
+			msg.addByte(0x01);
```

Comment:
multiple elements can be supported

structure:
u8 elements count

u8 element type
u16 +x element magic level
![obraz](https://user-images.githubusercontent.com/5205079/142780934-5989cdd7-e098-4c4f-a96d-e9df96386c3a.png)